src/libs/mps_comm/machine.cpp

```diff
@@ -0,0 +1,479 @@
+            std::this_thread::sleep_for(mock_ready_duration_);
+            mock_callback(OpcUtils::MPSRegister::STATUS_READY_IN, false);
+          }
+          return;
+        }
+				OpcUtils::MPSRegister registerOffset;
+				if (command < Station::STATION_BASE)
+					registerOffset = OpcUtils::MPSRegister::ACTION_ID_BASIC;
+				else
+					registerOffset = OpcUtils::MPSRegister::ACTION_ID_IN;
+
+				bool                  statusBit = (bool)(status & Status::STATUS_BUISY);
+				OpcUtils::MPSRegister reg;
+				reg = registerOffset + OpcUtils::MPSRegister::ACTION_ID_IN;
+				setNodeValue(registerNodes[reg], (uint16_t)command, reg);
+				reg = registerOffset + OpcUtils::MPSRegister::DATA_IN;
+				setNodeValue(registerNodes[reg].GetChildren()[0], (uint16_t)payload1, reg);
+				reg = registerOffset + OpcUtils::MPSRegister::DATA_IN;
+				setNodeValue(registerNodes[reg].GetChildren()[1], (uint16_t)payload2, reg);
+				reg = registerOffset + OpcUtils::MPSRegister::STATUS_ENABLE_IN;
+				setNodeValue(registerNodes[reg], statusBit, reg);
+				reg = registerOffset + OpcUtils::MPSRegister::ERROR_IN;
+				setNodeValue(registerNodes[reg], (uint8_t)error, reg);
+				logger->info("Finished command: {} {} {} {}", command, payload1, payload2, status);
+				success = true;
+			} catch (std::runtime_error &e) {
+				logger->warn("Failed to send command to {}, reconnecting", name_);
+				subscriptions.clear();
+				connect_PLC();
+				success = false;
+			}
+		} while (!success);
+	};
+	std::unique_lock<std::mutex> lock(command_queue_mutex_);
+	command_queue_.push(call);
+	lock.unlock();
+	queue_condition_.notify_one();
+	return;
+}
+
+void Machine::reset() {
+  send_command(machine_type_ | Command::COMMAND_RESET);
+}
+
+bool Machine::wait_for_ready(int timeout) {
+  struct timespec time_c, time_0, time_d;
+  if (timeout >= 0) {
+    clock_gettime(CLOCK_MONOTONIC, &time_0);
+  }
+    
+  while (!getReturnValue(OpcUtils::MPSRegister::STATUS_READY_IN)->ToInt())
+  {
+    {
+      std::lock_guard<std::mutex> l(lock_);
+      if (abort_operation_) {
+        std::cout << "Abort operation" << std::endl;
+        return false;
+      }
+    }
+    if (timeout >= 0) {
+      clock_gettime(CLOCK_MONOTONIC, &time_c);
+      timespec_diff( &time_0, &time_c, &time_d);
+      if( time_d.tv_sec * 1000 + time_d.tv_nsec / 1000000 > timeout*10) { /* TODO REMOVE *10 from timeout */
+        return false;
+      }
+    }
+  }
+  if (getReturnValue(OpcUtils::MPSRegister::STATUS_ENABLE_IN)->ToInt())
+    return true;
+  else
+    return false;
+}
+
+void Machine::wait_for_buisy() {
+  struct timespec time_c, time_0, time_d;
+  clock_gettime(CLOCK_MONOTONIC, &time_0);
+    
+  while ((getReturnValue(OpcUtils::MPSRegister::STATUS_BUSY_IN)->ToInt())) {
+    clock_gettime(CLOCK_MONOTONIC, &time_c);
+    timespec_diff( &time_0, &time_c, &time_d);
+    if( time_d.tv_sec * 1000 + time_d.tv_nsec / 1000000 > Timeout::TIMEOUT_BUISY*10) { /* TODO REMOVE *10 from timeout */
```

Comment:
```suggestion
    if( time_d.tv_sec * 1000 + time_d.tv_nsec / 1000000 > Timeout::TIMEOUT_BUSY*10) { /* TODO REMOVE *10 from timeout */
```
Another typo here i guess